bigbluebutton/greenlight
# Fix cascaded deletion of shared_access for users and rooms

## Description

Fixes #2444
Fixes #3054 

Deletion of `user`s and `room`s fails with a foreign key constraint violation when either are referenced in the `shared_access` table. This is resolved by marking `shared_access` for deletion in case of dependencies in the `user` and `room` models.

This occurs only when running against PostgreSQL, as default configurations of SQLite don't seem to enforce foreign keys. As the testsuite usually is executed against SQLite, we made sure that our tests also fail when deletions succeed and orphaned `shared_access` entities are left over.

This PR might be a candidate for squashing, for now, we created four separate commits for tests and fixes of both bugs.

## Testing Steps

* Checkout this branch without the last two commits, verify that two tests fail
* Checkout tip of this branch, verify that tests pass
